keras_nlp/src/utils/transformers/convert_bart.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+
+from keras_nlp.src.utils.preset_utils import HF_CONFIG_FILE
+from keras_nlp.src.utils.preset_utils import get_file
+from keras_nlp.src.utils.preset_utils import jax_memory_cleanup
+from keras_nlp.src.utils.preset_utils import load_config
+from keras_nlp.src.utils.transformers.safetensor_utils import SafetensorLoader
+
+
+def convert_backbone_config(transformers_config):
+    return {
+        "vocabulary_size": transformers_config["vocab_size"],
+        "num_layers": transformers_config["num_hidden_layers"],
+        "num_heads": transformers_config["encoder_attention_heads"],
+        "hidden_dim": transformers_config["d_model"],
+        "intermediate_dim": transformers_config["encoder_ffn_dim"],
+        "dropout": transformers_config["dropout"],
+        "max_sequence_length": transformers_config["max_position_embeddings"],
+    }
+
+
+def convert_weights(backbone, loader):
+    # Embeddings
+    loader.port_weight(
+        keras_variable=backbone.token_embedding.embeddings,
+        hf_weight_key="shared.weight",
+    )
+
+    # Encoder blocks
+    for index in range(backbone.num_layers):
+        encoder_layer = backbone.encoder_transformer_layers[index]
```

Comment:
very minor, but I think we could keep things under our 80 character limit if we did stuff like...

```
encoder_layer = backbone.encoder_transformer_layers[index]
self_attention = encoder_layer._self_attention_layer
hf_prefix = f"encoder.layers.{index}"

...

loader.port_weight(
    keras_variable=self_attention.query_dense.kernel,
    hf_weight_key=f"{hf_prefix}.self_attn.q_proj.weight",
    hook_fn=lambda hf_tensor, keras_shape: np.reshape(
        np.transpose(hf_tensor), keras_shape
    ),
)
```

Reply:
Done.